lib/embiggen/uri.rb

```diff
@@ -43,8 +43,17 @@ def follow(request_options = {})
       location = http_client.follow(timeout)
       fail BadShortenedURI.new(
         "following #{uri} did not redirect", uri) unless location
+      check_location(location)
 
       location
     end
+
+    def check_location(location)
+      addressable = Addressable::URI.parse(location)
+
+      fail Addressable::URI::InvalidURIError if addressable.host.nil?
```

Comment:
I wonder if we should invert this logic: check the correctness of a URL by looking for *at least* the following:

* A scheme (e.g. `http`);
* A host (e.g. `example.com`);
* A path (e.g. `/`).